#  THIS CODE DOCUMENT IS FOR MY MASTERS THESIS " 
Machine Learning in Football prediction for the Begian Jupiler League

In [1]:
# Importing usefull packages
import pandas as pd
import numpy as np
import matplotlib as plt

In [18]:

# Reading the datasets (assuming they are CSV files)
B1819 = pd.read_csv('B1819.csv')
B1920 = pd.read_csv('B1920.csv')
B2021 = pd.read_csv('B2021.csv')
B2122 = pd.read_csv('B2122.csv')
B2223 = pd.read_csv('B2223.csv')
B2324 = pd.read_csv('B2324.csv')



In [17]:
print(f'2018/2019 season shape: {B1819.shape}')
print(f'2019/2020 season shape: {B1920.shape}')
print(f'2020/2021 season shape: {B2021.shape}')
print(f'2021/2022 season shape: {B2122.shape}')
print(f'2022/2023 season shape: {B2223.shape}')
print(f'2023/2024 season shape: {B2324.shape}')

2018/2019 season shape: (240, 61)
2019/2020 season shape: (232, 105)
2020/2021 season shape: (306, 105)
2021/2022 season shape: (306, 106)
2022/2023 season shape: (306, 105)
2023/2024 season shape: (312, 105)


the data retrived from football.co.uk shows dissimilarities  accross the different years considered on the total number of matches played. 
-2018/2019 season had 240 games played 
-2019/2020 season had  232 games played
-2020/2021, 2021/2022, 2022/2023 all had 306 games played.
-2023/2024 season had 312 games played

**The differene in total number of games will be invegated and if need be will be adjusted.**

# Merging  datasets
 the 6 season datasets is combined into on main dataset  for further analyis and exploration.

In [6]:
# List of DataFrames
dfs = [B1819, B1920, B2021, B2122, B2223, B2324]

# Concatenate the DataFrames vertically (one below the other)
merged_data = pd.concat(dfs, ignore_index=True)


# Display the merged dataset
merged_data.head()

Div        Date    HomeTeam          AwayTeam  FTHG  FTAG FTR  HTHG  HTAG  \
0  B1  27/07/2018    Standard              Gent     3     2   H   1.0   1.0   
1  B1  28/07/2018    Kortrijk        Anderlecht     1     4   A   0.0   2.0   
2  B1  28/07/2018    Oostende          Mouscron     2     1   H   0.0   1.0   
3  B1  28/07/2018  St Truiden     Cercle Brugge     0     0   D   0.0   0.0   
4  B1  28/07/2018     Waregem  Waasland-Beveren     2     2   D   1.0   0.0   

  HTR  ...  AHCh  B365CAHH  B365CAHA  PCAHH  PCAHA  MaxCAHH  MaxCAHA  AvgCAHH  \
0   D  ...   NaN       NaN       NaN    NaN    NaN      NaN      NaN      NaN   
1   A  ...   NaN       NaN       NaN    NaN    NaN      NaN      NaN      NaN   
2   A  ...   NaN       NaN       NaN    NaN    NaN      NaN      NaN      NaN   
3   D  ...   NaN       NaN       NaN    NaN    NaN      NaN      NaN      NaN   
4   H  ...   NaN       NaN       NaN    NaN    NaN      NaN      NaN      NaN   

   AvgCAHA  Unnamed: 105  
0      NaN           NaN  
1      NaN           NaN  
2      NaN           NaN  
3      NaN           NaN  
4      NaN           NaN  

[5 rows x 126 columns]

In [8]:
merged_data.shape

(1702, 126)